extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java

```diff
@@ -179,16 +182,33 @@ private Response respond(List<ResourceDescription> descriptions) {
             }
             sb.resourceEnd();
         }
+
         if (descriptions.isEmpty()) {
             sb.noResourcesFound();
         }
+        sb.headerEnd();
+
+        if (!staticResources.isEmpty()) {
+            sb.header("Static Resources");
```

Comment:
You're using the header for something it shouldn't be used. That's why you have all those unrelated changes.